### PR TITLE
add .gitignore instead of .hgignore / bug reports now is linked to github

### PR DESCRIPTION
**Pull request by goanpeca**
The first PR for the new home :)

Thanks @ccordoba12

### CODE REVIEW

File: spyderlib/spyder.py [hunk 1]
```diff
@@ -1196,7 +1196,7 @@ def post_visible_setup(self):
             subprocess.call(['open', app_path + MAC_APP_NAME])
 
         # Server to maintain just one Spyder instance and open files in it if
-        # the user tries to start other instances with
+        # the user tries to start other instances w
```
Comment (Nodd) [inline spyderlib/spyder.py]: Uh, looks like an unwanted change here !

Also, hello everyone !

Comment (goanpeca) [inline spyderlib/spyder.py]: Sh*t!

Comment (Nodd) [inline spyderlib/spyder.py]: Now that we're using git you can hide your error by squashing the last three commit (then force push to update the PR) if you want :)

Comment (goanpeca) [inline spyderlib/spyder.py]: Yes, actually I wanted to do that, but I am not sure :)

Teach me how :p ?

**Comment by ccordoba12** [comment]
Gonzalo, for the next time you need to create a branch before starting to work on something with

    git branch

**Comment by goanpeca** [comment]
Oops... so, should I do it again... or ?

**Comment by Eric89GXL** [comment]
There are multiple methods for squashing commits, but I usually squash commits using:
```
git rebase -i HEAD~3
```
Then following the interactive prompts. You can change `pick` to `squash` or `s` for short, and it will squash upward. So you just `p` your first commit and `s` the ones below it. Then it should pop up asking what to call the commit message, and you can modify it.

**Before doing this**, it might be a good idea to work off of a different branch in case you make some terrible error. This is what I'd do:
```
$ git status
* master
$ git checkout -b squash
$ git status
  master
* squash
$ get rebase -i HEAD~3
<make the order p/s/s instead of p/p/p, fix commit message>
$ git push origin --force squash:master
```
And as pointed out, it would be better to work off of something other than master by default. I very recently made a standard GitHub workflow tutorial that might be helpful in case this stuff doesn't make sense:

https://github.com/LABSN/expyfun/blob/master/git_flow.rst

**Comment by goanpeca** [comment]
Thanks! I appreciate the information :)

**Comment by ccordoba12** [comment]
Do you plan to remove the bad commit? Or should I merge as it is?

**Comment by goanpeca** [comment]
Ok so I do not want to start this with the wrong foot so. I will close this, and restart and make these changes in 2 different pull request, because they are not really related.

ok? give me 5

**Comment by ccordoba12** [comment]
Yes, I think it's better

**Comment by goanpeca** [comment]
Thanks @Eric89GXL, I shamelessly copied and adapted your github workflow tutorial. Now is in the spyder wiki [here](https://github.com/spyder-ide/spyder/wiki/Github-Recommended-Workflow)

Should I add some disclaimer pointing to your original document, or thanking you directly :-)?

Cheers

**Comment by SylvainCorlay** [comment]
@goanpeca you could also have rebased with `-i` and skipped the commits. Unlike with hg, it is very easy to change history with git (I do it all the time for my PRs at work). Hence no need to close PRs.

**Comment by goanpeca** [comment]
Hi @SylvainCorlay, yes I learned that, but I was to lazy to do it at the moment, besides I did felt that they should have been separate PRs, one for the gitignore, and the other for the bug reports :)

**Comment by Eric89GXL** [comment]
@goanpeca I don't really care about a reference either way, but I usually do put a link to the original in documents like that. That way I remember to periodically check the original for updates/improvements to the original, or to open a PR into the original version with any sufficiently general improvements I make to the derivative work.

**Comment by goanpeca** [comment]
Good to know thanks!

**Pull request closed without merging**